activerecord/lib/active_record/skip_joins_association_relation.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+module ActiveRecord
+  class SkipJoinsAssociationRelation < Relation # :nodoc:
+    TOO_MANY_RECORDS = 5000
+
+    attr_reader :ids, :key
+
+    def initialize(klass, *args)
+      key, ids = args
+      @ids = ids.uniq
+      @key = key
+      super(klass)
+    end
+
+    def limit(value)
+      records.take(value)
+    end
+
+    def first(limit = nil)
+      if limit
+        records.limit(limit).first
+      else
+        records.first
+      end
+    end
+
+    def load
+      super
+      records = @records
+
+      if records.length > TOO_MANY_RECORDS
+        warn("You've requested to order #{records.length} in memory. This may have an impact on the performance of this query. Use with caution.")
```

Comment:
I doubt this warning will be useful. It will be buried in a lot of logs and depending in the configuration, since it is not going to the logger but to STDOUT it will just not be seeing. I think it is better to remove.